Runtime/CourseController/CourseControllerSetup.cs

```diff
@@ -149,5 +150,23 @@ public void ResetToDefault()
             Type courseControllerType = RetrieveDefaultControllerType();
             courseControllerQualifiedName = courseControllerType.Name;
         }
+
+#if UNITY_EDITOR
+        /// <summary>
+        /// Workaround to set the focus to the CourseControllerSetup if it has missing components that are required by the CourseController.
+        /// </summary>
+        [UnityEditor.Callbacks.DidReloadScripts]
+        private static void OnScriptsReloaded()
+        {
+            CourseControllerSetup setup = FindObjectOfType<CourseControllerSetup>();
+            List<Type> currentTypes = setup.GetComponents<Component>().Select(c => c.GetType()).ToList();
+            ICourseController cc = setup.GetCourseControllerFromType();
+            bool courseControllerHasMissingComponents = currentTypes.Except(cc.GetRequiredSetupComponents()).Any();
+            if (courseControllerHasMissingComponents)
+            {
+                Selection.activeObject = setup;
```

Comment:
Calling Setup here is probably the better way. But did not think about it before 🤔

Reply:
Setup does not work